Source/GameView/Repositories/GameAPIRepo.swift

```diff
@@ -0,0 +1,205 @@
+  public func getGameList(with serviceCaller: ServiceCaller) {
+
+    let url: URL = URL(string: SteamURLComponents.gameListURL)!
+    
+    // MARK: - Call Failed Closure
+    serviceCaller.callFailed = { error in
+      
+    }
+    
+    // MARK: - ---------------------------
+
+    // MARK: - Call Succeeded Closure
+    serviceCaller.callSucceeded = { [weak self] data, _ in
+
+      guard let self = self else {
+        return
+      }
+
+      let games = self.decodeGames(data: data)
+
+      var smallGameList: [Game] = []
+
+      //This limit exists until sectioning with the table works.
+      let size: Int = games.count >= 25 ? 25 : games.count
+
+      if games.count >= 1 {
+
+        for index in 0...(size - 1) {
+          smallGameList.append(games[index])
+        }
+
+        self.notifyGameListLoaded(gameList: smallGameList)
+
+      }
+
+    }
+    
+    // MARK: - ---------------------------
+    
+    do {
+      try serviceCaller.makeServiceCall(with: url, and: DataBundle())
+    } catch {
+      fatalError("Service call not set up properly")
+    }
+    
+  }
+  
+  public func getGameDetails(of gameList: [Game], with serviceCaller: ServiceCaller) {
+    
+    let dispatchGroup = DispatchGroup()
+    
+    gameList.forEach({ (game) in
+      
+      dispatchGroup.enter()
+      
+      let dataBundle = DataBundle()
+      dataBundle.extraData["game"] = game
+      
+      // MARK: - Call Succeeded Closure
+      serviceCaller.callSucceeded = { [weak self] data, dataBundle in
+        
+        guard let game = dataBundle.extraData["game"] as? Game else {
+          return
+        }
+        
+        guard let self = self else {
+          return
+        }
+        
+       self.decodeGameDetails(using: data, basedOn: game)
+        
+      }
+      
+      // MARK: - ---------------------------
+
+      // MARK: - Call Failed Closure
+      serviceCaller.callFailed = { error in
+        
+      }
+      
+      // MARK: - ---------------------------
+      
+      let urlString = "\(SteamURLComponents.specificGameURL)\(game.appID)"
+      let url: URL = URL(string: urlString)!
+      
+      do {
+        try serviceCaller.makeServiceCall(with: url, and: dataBundle)
+      } catch {
+        fatalError("Service call not set up properly")
+      }
+      
+      dispatchGroup.leave()
+      
+    })
+    
+    dispatchGroup.wait()
+    
+    self.notifyGameDetailsLoaded(withData: gameList)
+    
+  }
+  
+  private func decodeGames(data: Data) -> [Game] {
+    
+    var games: [Game] = []
+    
+    if let decodedGames = try? JSONDecoder().decode(AppList.self, from: data) {
+
+      decodedGames.applist.apps.forEach { (jsonGame) in
+        games.append(Game(appid: "\(jsonGame.appid)", name: jsonGame.name))
+      }
+
+    }
+    
+    return games
+    
+  }
+  
+  private func extractGameData(detailsWrapper: [String: Any], game: Game) {
+
+    guard let gameDetailsData = detailsWrapper["data"] as? [String: Any] else {
```

Comment:
We can probably save these keys into a struct instead of calling them with Strings